### PR TITLE
[IOPLT-571] Implement get activation job

### PR DESCRIPTION
**Pull request by kin0992**
#### List of Changes

- Implement the handler to get an activation job
- Create the ActivationJobReader
    - Implement the `get` operation to retrieve the activation job

#### Motivation and Context

We need a way to get the information about an activation job.

#### How Has This Been Tested?

Unit tests.

#### Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] I have committed only changes relevant to the task.
- [x] I have minimized the number of changes.
- [x] My changes are about only one task or issue.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/functions-subscription/src/adapters/azure/cosmosdb/__tests__/activation-job.test.ts [hunk 1]
```diff
@@ -1,13 +1,35 @@
...
+      )();
+
+      expect(actual).toStrictEqual(E.right(O.some(anActivationJob)));
+      expect(mockDB.container('').item).toBeCalledWith(id, trialId);
+    });
+  });
+
   describe('insert', () => {
-    it('should insert the item witout error', async () => {
+    it('should insert the item without error', async () => {
```
Comment (kin0992) [inline apps/functions-subscription/src/adapters/azure/cosmosdb/__tests__/activation-job.test.ts:32]: **note**: just fixed a typo

File: apps/functions-subscription/src/adapters/azure/functions/__tests__/get-activation-job.test.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    );
+
+    const expectedJson = {
+      usersToActivate: anActivationJob.usersToActivate,
+      usersActivated: anActivationJob.usersActivated,
+      trialId: anActivationJob.trialId,
+    };
+
+    expect(actual.status).toStrictEqual(200);
+    expect(await actual.json()).toMatchObject(expectedJson);
```
Comment (datalek) [inline apps/functions-subscription/src/adapters/azure/functions/__tests__/get-activation-job.test.ts:27]: **question**: Given that the compiler will produce a type error if the structure does not match, do you think this check makes the code more robust?

File: apps/functions-subscription/src/adapters/azure/functions/__tests__/get-activation-job.test.ts [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { describe, expect, it } from 'vitest';
+import * as TE from 'fp-ts/lib/TaskEither';
+import * as O from 'fp-ts/lib/Option';
+import { makeFunctionContext, makeTestSystemEnv } from './mocks';
+import { makeAValidGetActivationJobRequest } from './data';
+import { anActivationJob } from '../../../../domain/__tests__/data';
+import { makeGetActivationJobHandler } from '../get-activation-job';
+
+describe('getActivationJob', () => {
```
Comment (kin0992) [inline apps/functions-subscription/src/adapters/azure/functions/__tests__/get-activation-job.test.ts:9]:
```suggestion
describe('makeGetActivationJobHandler', () => {
```

File: apps/functions-subscription/src/adapters/azure/cosmosdb/activation-job.ts [hunk 4]
```diff
@@ -2,14 +2,24 @@ import { pipe } from 'fp-ts/lib/function';
...
   db: Database,
-): ActivationJobWriter => {
+): ActivationJobReader & ActivationJobWriter => {
   const container = db.container('activations');
   return {
+    get: (id) =>
+      pipe(
+        TE.tryCatch(() => container.item(id, id).read(), E.toError),
+        TE.flatMapEither(decodeFromItem(ActivationJobCodec)),
+      ),
```
Comment (datalek) [inline apps/functions-subscription/src/adapters/azure/cosmosdb/activation-job.ts:22]: **issue**: I'm not sure if this cause any issues.

```suggestion
        TE.flatMapEither(decodeFromItem(ActivationJobCodec)),
        TE.mapLeft(cosmosErrorToDomainError),
      ),
```

**Review by datalek — CHANGES_REQUESTED** [review]

**Review by datalek — APPROVED** [review]
LGTM

**Pull request merged**